flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/NetworkBuffer.java

```diff
@@ -639,4 +632,16 @@ public boolean isCompressed() {
 	public void setCompressed(boolean isCompressed) {
 		this.isCompressed = isCompressed;
 	}
+
+	@Override
+	public DataType getDataType() {
+		return dataType;
+	}
+
+	@Override
+	public void setDataType(DataType dataType) {
+		ensureAccessible();
```

Comment:
Is it necessary to `ensureAccessible` for setter?

Reply:
I copied this from tagAsEvent, IMO, ensureAccessible is not a must here and I checked the code, ensureAccessible method mainly checks about the reference count and ensures that the buffer is not recycled, so I  think it should be ok to keep this check.